### DISCUSSION TITLE
Alternative rendering backends

### DISCUSSION THREAD

**Discussion by skreborn** [sycamore-rs/sycamore#758] [General]
Now that Sycamore 0.9 is out with Reactivity v3 and View v2, I was wondering if we've gotten closer to or further from a future where alternative rendering backends (be it a TUI, GTK, or anything else) are a possibility.

**Comment by lukechu10** [comment]
Sycamore v0.9 makes the view API much more closely coupled with the HTML backend. I think this is a good thing. The previous architecture where we tried to keep the view API as decoupled as possible from the rendering backend meant that we had to make a lot of compromises in terms of API ergonomics, just for the sake for a little bit more of code reuse between backends.

This means that while the new architecture makes the view more tightly coupled with the web, I think this will make it easier to add alternative backends since we no longer have to work around the limitations of a web-first API. This also means that when creating an alternative rendering backend, there will need to be some duplicate work in implementing certain features such as list-diffing, etc. However, often times these things are also somewhat platform dependent anyways so it's not that much extra work.

Right now, I'm more focused on getting some remaining things done and hopefully release a 0.9.1 version soon. I'm definitely interested, however, in exploring new rendering backends for Sycamore and is a tentative goal for version 0.10.x

**Reply by skreborn** [reply]
Thank you for the detailed response! I'm honestly most interested in TUIs as a target as the currently available options in the space are quite lacking when it comes to fine-grained reactivity.

I understand and actually agree that a little bit of duplicate work is a perfectly good trade-off for the opportunity to design familiar, yet custom view code for alternative backends.

I'm anxiously awaiting progress, and I'll try to familiarize myself with the codebase in the meantime.
